Yahya w1 browsers

YahyaGanjo opened the pull request.

wilgert commented on 2-Browsers/Week1/homework/ex1-bookList/index.js:
```diff
@@ -40,7 +40,51 @@ const myBooks = [
+  booksList.style.listStyle = 'none';
+  booksList.style.display = 'flex';
+  titleAuthor.forEach((paragraph) => {
+    const listItems = document.createElement('li');
```
small thing: `listItems` is actually a single item. So it should be called `listItem`

wilgert commented on 2-Browsers/Week1/homework/ex1-bookList/index.js:
```diff
@@ -40,7 +40,51 @@ const myBooks = [
+
+  // set the background color of each list item
+
+  for (let i = 0; i < listItems.length; i++) {
```
Do you think it is possible to add the color without looping of the items again?

YahyaGanjo replied:
actually, using forEach is the closest thing that came to my head

wilgert commented on 2-Browsers/Week1/homework/ex2-aboutMe/index.js:
```diff
@@ -8,4 +8,17 @@
+
+// Adding class `list-item` to li
+const lis = Array.from(document.querySelectorAll('li'));
+lis.forEach((li) => (li.className = 'list-item'));
```
Good job!

wilgert commented on 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js:
```diff
@@ -6,7 +6,14 @@
+  document.body.textContent = currentTime;
+  document.body.style.textAlign = 'center';
+  // keeping time updated
+  setInterval(addCurrentTime, 1000);
```
Have you tried running this in the browser, and keeping the browser open for a while? It will crash your browser! Can you explain why that happens.

YahyaGanjo replied:
Is it because the setInterval has been called inside the function itself, which will duplicate the function every single loop causing StackOverflow

wilgert replied:
Exactly!

wilgert commented on 2-Browsers/Week1/homework/ex5-catWalk/index.js:
```diff
@@ -20,7 +20,36 @@
+    }
+
+    // switch to dance cat then back to the original
+    if (position === document.body.clientWidth / 2) {
```
On my screen the cat is not stopping in the middle. Can you figure out why?

YahyaGanjo replied:
yes its because position is being incremented 10 px by once which makes the equality invisible and false forever for some screen sizes.

wilgert commented on 2-Browsers/Week1/homework/ex3-hijackLogo.js:
```diff
@@ -7,6 +7,9 @@
 ------------------------------------------------------------------------------*/
 function hijackGoogleLogo() {
   // TODO your code goes in here
+  const logo = document.getElementsByClassName('lnXdpd');
```
This className `lnXdpd` looks auto-generated and it could change whenever google updates their homepage. Could you write a css selector that is more likely to keep working when google changes their page?

wilgert commented on 2-Browsers/Week1/homework/ex3-hijackLogo.js:
```diff
@@ -7,6 +7,9 @@
+  logo.srcset = 'https://avatars.githubusercontent.com/u/20858568?s=200&v=4';
 }
 
 hijackGoogleLogo();
```
Did you test this code, apart from the className issue I mentioned above it is not working because of a fundamental bug. Can you figure out what it is?

YahyaGanjo replied:
I must be exhausted when I wrote this code, getElementsByClassName returns HTML collection, so adding src to logo is simply nonsense.

wilgert reviewed (changes requested):
You did quite well, I have added some questions to help you learn. Exercise 3 and 5 don't work yet and you need to fix those!

YahyaGanjo commented:
> You did quite well, I have added some questions to help you learn. Exercise 3 and 5 don't work yet and you need to fix those!

Thank you very much for the feedback, searching answers to your questions makes the information memorable,
I wish I implemented your feedback properly
